src/wirecloud/catalogue/search_indexes.py

```diff
@@ -0,0 +1,198 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2017 CoNWeT Lab., Universidad Politécnica de Madrid
+
+# This file is part of Wirecloud.
+
+# Wirecloud is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# Wirecloud is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+
+# You should have received a copy of the GNU Affero General Public License
+# along with Wirecloud.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from six.moves.urllib.parse import urlparse, urljoin
+
+from django.db.models import Q
+
+from wirecloud.catalogue.models import CatalogueResource, get_template_url
+from wirecloud.commons.search_indexes import buildSearchResults, SearchQuerySet
+from wirecloud.commons.utils.version import Version
+from wirecloud.commons.utils.http import get_absolute_reverse_url
+
+from haystack import indexes
+
+
+class CatalogueResourceIndex(indexes.SearchIndex, indexes.Indexable):
+
+    text = indexes.CharField(document=True)
+
+    from wirecloud.commons.haystack_backends.solr_backend import GroupedSolrSearchBackend
```

Comment:
You are right, it should be at the top.

Currently, its there as a placeholder while we look for a better solution that does not require to use a different field depending on the searching back-end.